src/pages/index.js

```diff
@@ -1,155 +1,319 @@
+  return cardElement;
+}
+
+function handleDelete(cardId, cardElement) {
+  handleConfirmationOpen(cardId, cardElement);
 }
 
-// Initialize card section
+const deletePopup = new PopupWithConfirm(
+  "#verify__card__delete",
+  (cardId, cardElement) => {
+    api
+      .deleteCard(cardId)
+      .then(() => {
+        cardElement.remove();
+        deletePopup.close();
+        existingCardIds.delete(cardId); // Remove from Set when card is deleted
+      })
+      .catch((error) => {
+        console.error("Error deleting card:", error);
+      });
+  }
+);
+
 const cardSection = new Section(
   {
-    items: initialCards,
-    renderer: (data) => {
-      // Create a new card
-      const cardElement = renderCard(data);
-
-      // Display each card
-      cardSection.addItem(cardElement);
+    items: [],
+    renderer: (item) => {
+      const cardElement = createCard(item);
+      if (cardElement) {
+        cardSection.addItem(cardElement);
+      }
     },
   },
-  formSettings.cardList
+  ".cards__list"
 );
 
-// Enable validation for all forms
+const addCardPopup = new PopupWithForm("#modalAddCard", (data) => {
+  addCardPopup.showLoading(); // Show loading before API call
+  api
+    .renderCard(data.name, data.link)
+    .then((newCardData) => {
+      const cardElement = createCard(newCardData);
+      if (cardElement) {
+        cardSection.addItem(cardElement);
 
-/* -------------------------------------------------------------------------- */
-/*                                  Add Crad  Form                            */
-/* -------------------------------------------------------------------------- */
-
-// Create the add form instance
-const addCardPopup = new PopupWithForm(formSettings.modalAddCard, (data) => {
-  // Create a new card
-  const newCard = renderCard(data);
-  // Close the add form
-  addCardPopup.close();
-  // Add the new card to the section
-  cardSection.addItem(newCard);
+        const formName = document
+          .querySelector("#modalAddCard .modal__form")
+          .getAttribute("name");
+        formValidator[formName].resetValidation();
+        addCardPopup.close();
+      }
+    })
+    .catch((error) => {
+      console.error("Error creating new card:", error);
+    })
+    .finally(() => {
+      addCardPopup.hideLoading();
+    });
 });
-
-// Open the modal when users click on the add button
-addCardBut.addEventListener("click", () => {
-  // Open the add card form
+addCardButton.addEventListener("click", () => {
   addCardPopup.open();
-  formValidator[addCardForm.getAttribute("name")].resetValidation();
 });
 
-// Set add form event listeners
-addCardPopup.setEventListeners();
+api
+  .showPromiseStatus()
+  .then(({ initialCards, fetchedUserInfo }) => {
+    userInfo.setUserInfo(fetchedUserInfo);
+    userInfo.setAvatarInfo(fetchedUserInfo);
+    initialCards.forEach((card) => {
+      const cardElement = createCard(card);
+      if (cardElement) {
+        cardSection.addItem(cardElement);
```

Comment:
Please, delete ` initialCards.forEach((card) => {` because you render the cards with `cardSection.renderItems(initialCards);`